Source/CarthageKit/GitHub.swift

```diff
@@ -64,55 +64,142 @@ extension GitHubError: Decodable {
 	}
 }
 
-/// Describes a GitHub.com repository.
+/// Describes a GitHub.com or GitHub Enterprise repository.
 public struct GitHubRepository: Equatable {
+
+	/// Represents a URL for a GitHub website.
+	public enum URL: Equatable, Hashable, Printable {
+		/// The GitHub.com domain.
+		case GitHub
+
+		/// A GitHub Enterprise instance with its hostname.
+		case Enterprise(scheme: String, hostname: String)
+
+		public var scheme: String {
+			switch self {
+			case .GitHub:
+				return "https"
+
+			case let .Enterprise(scheme, _):
+				return scheme
+			}
+		}
+
+		public var hostname: String {
+			switch self {
+			case .GitHub:
+				return "github.com"
+
+			case let .Enterprise(_, hostname):
+				return hostname
+			}
+		}
+
+		public var APIEndpoint: String {
+			switch self {
+			case .GitHub:
+				return "\(scheme)://api.\(hostname)"
+
+			case .Enterprise:
+				return "\(description)/api/v3"
+			}
+		}
+
+		public var hashValue: Int {
+			return scheme.hashValue ^ hostname.hashValue
+		}
+
+		public var description: String {
+			return "\(scheme)://\(hostname)"
+		}
+	}
+
+	public let baseURL: URL
 	public let owner: String
 	public let name: String
 
 	/// The URL that should be used for cloning this repository over HTTPS.
 	public var HTTPSURL: GitURL {
-		return GitURL("https://github.com/\(owner)/\(name).git")
+		return GitURL("\(baseURL)/\(owner)/\(name).git")
 	}
 
 	/// The URL that should be used for cloning this repository over SSH.
 	public var SSHURL: GitURL {
-		return GitURL("ssh://git@github.com/\(owner)/\(name).git")
+		return GitURL("ssh://git@\(baseURL.hostname)/\(owner)/\(name).git")
 	}
 
 	/// The URL for filing a new GitHub issue for this repository.
 	public var newIssueURL: NSURL {
-		return NSURL(string: "https://github.com/\(owner)/\(name)/issues/new")!
+		return NSURL(string: "\(baseURL)/\(owner)/\(name)/issues/new")!
 	}
 
-	public init(owner: String, name: String) {
+	public init(baseURL: URL = .GitHub, owner: String, name: String) {
+		self.baseURL = baseURL
 		self.owner = owner
 		self.name = name
 	}
 
 	/// Parses repository information out of a string of the form "owner/name".
 	public static func fromNWO(NWO: String) -> Result<GitHubRepository, CarthageError> {
-		let components = split(NWO, maxSplit: 1, allowEmptySlices: false) { $0 == "/" }
-		if components.count < 2 {
-			return .failure(CarthageError.ParseError(description: "invalid GitHub repository name \"\(NWO)\""))
+		// GitHub.com
+		let components = split(NWO, maxSplit: 2, allowEmptySlices: false) { $0 == "/" }
+		if components.count == 2 {
+			return .success(self(owner: components[0], name: components[1]))
+		}
+
+		// GitHub Enterprise
+		if let
+			URL = NSURL(string: NWO),
```

Comment:
`NWO` stands for _Name With Owner_. Since this method now takes a GitHub Enterprise URL, it's not really appropriate terminology. Something like `fromIdentifier` or `fromSpecifier` seems more appropriate.